model/shared/src/main/scala/explore/CallForProposal.scala

```diff
@@ -36,13 +37,24 @@ case class CallForProposal(
 ) derives Eq,
       Decoder:
 
-  def deadline(proposalPartners: List[Partner]): Option[Timestamp] =
-    val callPartners = partners
-    proposalPartners
-      .map(p => callPartners.find(_.partner === p).flatMap(_.submissionDeadline))
-      .minimumOption
-      .flatten
-      .orElse(submissionDeadlineDefault)
+  def deadline(piPartner: Option[PartnerLink]): Option[Timestamp] =
+    piPartner.flatMap {
```

Comment:
Maybe this logic should go to core

Reply:
So I think you can ignore / delete the `submissionDeadlineDefault` unless you're making an editor for the CfP.  Just ask each partner for its deadline and it will handle it. You can also add the `nonPartnerDeadline` and get it from the query instead of computing it.